src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -35,6 +35,15 @@ class MockHttpClient implements HttpClientInterface
      * @param callable|callable[]|ResponseInterface|ResponseInterface[]|iterable|null $responseFactory
      */
     public function __construct($responseFactory = null, ?string $baseUri = 'https://example.com')
+    {
+        $this->setResponseFactory($responseFactory);
+        $this->defaultOptions['base_uri'] = $baseUri;
+    }
+
+    /**
+     * @param callable|callable[]|ResponseInterface|ResponseInterface[]|iterable|null $responseFactory
+     */
+    public function setResponseFactory($responseFactory)
```

Comment:
I think you should add `: void` as return type

Reply:
oh, or better, make this a wither: `withResponseFactory(): static`?

Reply:
How `wither` would help in unit tests when you instantiated a mock client in a `setUp` method and you need to change the factory in a specific test?

Reply:
good point :)

Reply:
Added return type.